### DISCUSSION TITLE
Any examples of how to use the "vehicle envelopes"?

### DISCUSSION THREAD

**Discussion by Dknife7** [openTCS/opentcs#128] [Q&A]
we been running into the deadlock situation and noticed the new "vehicle envelope" function updated in 5.11 version might be helpful. Has anyone tried to utilize that in any project? We are a bit clueless about how to set that up. Any hints will be helpful

**Comment by swltr** [comment]
The model editor application does not support vehicle envelopes, yet, and it is definitely underdocumented at this point. You can only set them via the Java API or the web API right now. (Basically, an envelope is a polygon describing the area covered by a vehicle on a path/point.)

**Comment by Dknife7** [comment]
Thank you for replying. What are some use cases of envelope. Have you tried it in any project？

**Reply by swltr** [reply]
Envelopes, if set in a plant model, are intended to be used to avoid collisions between vehicles. Yes, this feature, like most others in openTCS, has been implemented explicitly for use in a real project.
